File: tools/write_architect.lua
return {
    name = "write_architect",
    description = 'Write architecture to ARCHITECT.md. Input should be in JSON format: {"content": "file content"}',
    tool = {
        type = "function",
        ["function"] = {
            name = "write_architect",
            description = 'Write architecture to ARCHITECT.md. Input should be in JSON format: {"content": "file content"}',
            parameters = {
                type = "object",
                properties = {
                    content = {
                        type = "string",
                        description = "The content to write to the ARCHITECT.md"
                    }
                },
                required = {"content"}
            }
        }
    },
    action = function(input)
        local file_name = "ARCHITECT.md"
        local content = input.content
        print("\\e[0;36mWriting file: ")
        print(content)
        print("-> will be written to: " .. file_name)
        if not (AskProceed("write_file")) then
            return { success = false, message = "Tool calling denied by user" }
        end
        local file, errmsg = io.open(file_name, "w")
        if file then
            file:write(content)
            file:close()
            return { success = true, content = "File written successfully" }
        else
            return { success = false, content = "Failed to write file: " .. errmsg }
        end
    end
}

File: tools/write_architect.py
"""write_architect tool — write ARCHITECT.md design document."""

name = "write_architect"
description = 'Write architecture to ARCHITECT.md. Input: {"content": "..."}'
schema = {
    "type": "function",
    "function": {
        "name": "write_architect",
        "description": 'Write architecture to ARCHITECT.md. Input: {"content": "..."}',
        "parameters": {
            "type": "object",
            "properties": {
                "content": {"type": "string", "description": "The content to write to ARCHITECT.md"},
            },
            "required": ["content"],
        },
    },
}


def action(input: dict) -> dict:
    from communication import ask_proceed

    content = input.get("content", "")
    print(f"\033[0;36mWriting ARCHITECT.md:")
    print(content)
    if not ask_proceed("write_file"):
        return {"success": False, "message": "Tool calling denied by user"}
    try:
        with open("ARCHITECT.md", "w") as f:
            f.write(content)
        return {"success": True, "content": "File written successfully"}
    except Exception as e:
        return {"success": False, "content": f"Failed to write file: {e}"}
